### ISSUE TITLE
`candidateBase` conn member race condition

### ISSUE THREAD

**Issue by Sean-Der** [pion/ice#242]
From `pion/webrtc` test run

```
==================
WARNING: DATA RACE
Write at 0x00c00029ea68 by goroutine 115:
  github.com/pion/ice/v2.(*candidateBase).close()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/candidate_base.go:178 +0x104
  github.com/pion/ice/v2.(*CandidateHost).close()
      <autogenerated>:1 +0x43
  github.com/pion/ice/v2.(*Agent).deleteAllCandidates()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/agent.go:887 +0x10a
  github.com/pion/ice/v2.(*Agent).taskLoop.func2()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/agent.go:196 +0x3c
  github.com/pion/ice/v2.(*Agent).taskLoop()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/agent.go:216 +0x24f
Previous read at 0x00c00029ea68 by goroutine 12:
  github.com/pion/ice/v2.(*candidateBase).writeTo()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/candidate_base.go:185 +0x7a
  github.com/pion/ice/v2.(*CandidateHost).writeTo()
      <autogenerated>:1 +0x89
  github.com/pion/ice/v2.(*Conn).Write()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/candidatepair.go:86 +0x2d0
  github.com/pion/webrtc/v3/internal/mux.(*Endpoint).Write()
      /home/travis/gopath/src/github.com/pion/webrtc/internal/mux/endpoint.go:41 +0xb0
  github.com/pion/dtls/v2/internal/net/connctx.(*connCtx).Write()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/internal/net/connctx/connctx.go:121 +0x274
  github.com/pion/dtls/v2.(*Conn).writePackets()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:396 +0xb8f
  github.com/pion/dtls/v2.(*handshakeFSM).send()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/handshaker.go:223 +0x9a
  github.com/pion/dtls/v2.(*handshakeFSM).Run()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/handshaker.go:156 +0x3e9
  github.com/pion/dtls/v2.(*Conn).handshake.func2()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:807 +0xe6
Goroutine 115 (running) created at:
  github.com/pion/ice/v2.NewAgent()
      /home/travis/gopath/pkg/mod/github.com/pion/ice/v2@v2.0.0-rc.4/agent.go:339 +0x109a
  github.com/pion/webrtc/v3.(*ICEGatherer).createAgent()
      /home/travis/gopath/src/github.com/pion/webrtc/icegatherer.go:123 +0xdd0
  github.com/pion/webrtc/v3.(*ICETransport).ensureGatherer()
      /home/travis/gopath/src/github.com/pion/webrtc/icetransport.go:292 +0x116
  github.com/pion/webrtc/v3.(*ICETransport).AddRemoteCandidate()
      /home/travis/gopath/src/github.com/pion/webrtc/icetransport.go:252 +0xcc
  github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:834 +0x6ea
  github.com/pion/webrtc/v3.TestPeerConnection_Close_DuringICE()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection_close_test.go:141 +0x4f6
  testing.tRunner()
      /home/travis/.gimme/versions/go1.13.linux.amd64/src/testing/testing.go:909 +0x199
Goroutine 12 (running) created at:
  github.com/pion/dtls/v2.(*Conn).handshake()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:805 +0x4e2
  github.com/pion/dtls/v2.createConn()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:194 +0x155a
  github.com/pion/dtls/v2.ClientWithContext()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:251 +0x103
  github.com/pion/dtls/v2.Client()
      /home/travis/gopath/pkg/mod/github.com/pion/dtls/v2@v2.0.1/conn.go:220 +0xe8
  github.com/pion/webrtc/v3.(*DTLSTransport).Start()
      /home/travis/gopath/src/github.com/pion/webrtc/dtlstransport.go:312 +0x27a
  github.com/pion/webrtc/v3.(*PeerConnection).startTransports()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:1676 +0x3d2
  github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription.func2()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:871 +0x19a
  github.com/pion/webrtc/v3.(*operations).start()
      /home/travis/gopath/src/github.com/pion/webrtc/operations.go:77 +0xce
==================
```

**Comment by jech** [comment]
I think this was fixed in e8ac36354ccb2307134e25a83c4aeb514f5ed160.

**Comment by Sean-Der** [comment]
:+1: you are right, thanks @jech!
